### ISSUE TITLE
vParquet3: Configurable dedicated columns for span/resource attributes

### ISSUE THREAD

**Issue by stoewer** [grafana/tempo#2527]
This issue bundles tasks related to extending Tempo’s Parquet based block format such that a configurable subset of span attributes can be stored in dedicated columns. The goal is to improve the search performance of TraceQL in Tempo.

The extended format will be released as `vParquet3`.

### Context

* [Design proposal PR](https://github.com/grafana/tempo/pull/2525) (replace with link to doc once merged)

```[tasklist]
### Planning
- [x] Create draft for design proposal
- [x] Set up development main branch ([`main-vparquet3`](https://github.com/grafana/tempo/tree/main-vparquet3))
- [x] Publish design proposal to Tempo (grafana/tempo#2525)
```

```[tasklist]
### Implementation
- [ ] https://github.com/grafana/tempo/issues/2528
- [ ] https://github.com/grafana/tempo/issues/2548
- [ ] https://github.com/grafana/tempo/issues/2552
- [ ] https://github.com/grafana/tempo/issues/2570
- [ ] https://github.com/grafana/tempo/issues/2576
- [ ] https://github.com/grafana/tempo/issues/2630
- [ ] https://github.com/grafana/tempo/issues/2663
- [ ] https://github.com/grafana/tempo/issues/2629
- [ ] https://github.com/grafana/tempo/issues/2610
- [ ] https://github.com/grafana/tempo/issues/2674
```

```[tasklist]
### Release
- [x] Create and release vParquet3 encoding from the development branch
- [x] Create dashboard to track TraceQL performance improvements
- [x] Rollout to ops and test configs
- [x] Rollout to production cells
```

**Comment by knylander-grafana** [comment]
FYI: Since vParquet3 wasn't included as part of release 2.2, I've rolled back some of the doc changes. They are still in the files, but are commented out. 
* operations/dedicated_columns.md is set to not publish
* Analyze blocks and analyze block sections of the API doc have the sentences referencing v3 commented out.

**Comment by knylander-grafana** [comment]
Doc issue for dedicated columns: https://github.com/grafana/tempo-squad/issues/293
